### DISCUSSION TITLE
Merge two PNG files with size adjustment

### DISCUSSION THREAD

**Discussion by abilask** [ImageMagick/ImageMagick#5069] [Help] [answered]
I need Merge two PNG files (file size may vary) with size adjustment, here the file size need to be adjusted by reading during execution.
Example expected output is given. Please help with the program.
11111.png + 22222.png = 33333.png
![11111](https://user-images.githubusercontent.com/90464818/164911794-922dbf95-4cb1-4309-8596-df1db4475159.png)
![22222](https://user-images.githubusercontent.com/90464818/164911797-933f17c3-b254-4303-bea6-6bebdb2bffcf.jpg)
![33333](https://user-images.githubusercontent.com/90464818/164911799-4855ea9f-4e42-4ed6-9491-ef9b230d811d.png)

**Comment by fmw42** [comment]
What is your ImageMagick version and platform/OS?

Does  this need to fully automatic scaling to fit or will you specify the size reduction of the image so that it will fit into the frame?

**Reply by abilask** [reply]
ImageMagick-7.1.0-30 on Linux (Ubuntu)

**Reply by abilask** [reply]
Size I mean input image 22222.png is fixed size, but the 11111.png example , i may use another image of different size but the out put of the png is with equal gap with the line of 22222.png image, I hope i have given my requirement clearly  . thanks.

**Comment by fmw42** [comment, marked as answer]
Here is one way using bash unix scripting. I do a virtual -trim to get the size of black frame in the frame image. I set the amount of padding inside the black frame for the size of the image and compute the size and position. Then I use that size and offset to do the compositing.

```
trim=`magick frame.jpg -format "%@" info: | tr "x" "+"`
ww=`echo $trim | cut -d+ -f1`
hh=`echo $trim | cut -d+ -f2`
xo=`echo $trim | cut -d+ -f3`
yo=`echo $trim | cut -d+ -f4`
padx=90
pady=60
ww=$((ww-2*padx))
hh=$((hh-2*pady))
xo=$((xo+padx))
yo=$((yo+pady))
magick frame.jpg \( photo.png -resize ${ww}x${hh} \) -geometry +${xo}+${yo} -compose over -composite photo_in_frame.jpg

```

![photo_in_frame](https://user-images.githubusercontent.com/13458218/164912926-f893acad-538d-4600-879c-f94f9d3097e6.jpg)

EDIT:

Here is a pure IM 7 in-line computation solution:

```
cd
cd desktop
padx=90
pady=60
magick frame.jpg -write mpr:frame -fuzz 0 -trim \
-set option:ww "%[fx:w-2*$padx]" -set option:hh "%[fx:h-2*$pady]" \
-set option:xo "%[fx:page.x+$padx]" -set option:yo "%[fx:page.y+$pady]" +repage -delete 0 \
mpr:frame \( photo.png -resize "%[ww]x%[hh]" \) -geometry "+%[xo]+%[yo]" -compose over -composite photo_in_frame.jpg

```

**Comment by abilask** [comment]
Thanks Fred. 

Sent from Yahoo Mail on Android 
 
  On Sat, 23 Apr 2022 at 9:01 pm, Fred ***@***.***> wrote:   


Here is one way using bash unix scripting.
trim=`magick frame.jpg -format "%@" info: | tr "x" "+"`
ww=`echo $trim | cut -d+ -f1`
hh=`echo $trim | cut -d+ -f2`
xo=`echo $trim | cut -d+ -f3`
yo=`echo $trim | cut -d+ -f4`
padx=90
pady=60
ww=$((ww-2*padx))
hh=$((hh-2*pady))
xo=$((xo+padx))
yo=$((yo+pady))
magick frame.jpg \( photo.png -resize ${ww}x${hh} \) -geometry +${xo}+${yo} -compose over -composite photo_in_frame.jpg
